framework/packages/abstract-std/src/objects/ownership/gov_ownable.rs

```diff
@@ -0,0 +1,709 @@
+            ownership.assert_can_change_ownership()?;
+
+            // Assert admin
+            check_owner(querier, ownership, sender)?;
+        }
+    }
+
+    Ok(())
+}
+
+/// Update the contract's ownership info based on the given action.
+/// Return the updated ownership.
+pub fn update_ownership(
+    deps: DepsMut,
+    block: &BlockInfo,
+    sender: &Addr,
+    version_control: Addr,
+    action: GovAction,
+) -> Result<Ownership<Addr>, GovOwnershipError> {
+    match action {
+        GovAction::TransferOwnership { new_owner, expiry } => {
+            transfer_ownership(deps, sender, new_owner, version_control, expiry)
+        }
+        GovAction::AcceptOwnership => accept_ownership(deps.storage, &deps.querier, block, sender),
+        GovAction::RenounceOwnership => renounce_ownership(deps.storage, &deps.querier, sender),
+    }
+}
+
+/// Get the current ownership value.
+pub fn get_ownership(storage: &dyn Storage) -> StdResult<Ownership<Addr>> {
+    OWNERSHIP.load(storage)
+}
+
+pub fn query_ownership<Q: CustomQuery>(
+    querier: &QuerierWrapper<Q>,
+    remote_contract: Addr,
+) -> StdResult<Ownership<Addr>> {
+    OWNERSHIP.query(querier, remote_contract)
+}
+
+/// Propose to transfer the contract's ownership to the given address, with an
+/// optional deadline.
+fn transfer_ownership(
+    deps: DepsMut,
+    sender: &Addr,
+    new_owner: GovernanceDetails<String>,
+    version_control: Addr,
+    expiry: Option<Expiration>,
+) -> Result<Ownership<Addr>, GovOwnershipError> {
+    let new_owner = new_owner.verify(deps.as_ref(), version_control)?;
+
+    if new_owner.owner_address(&deps.querier).is_none() {
+        return Err(GovOwnershipError::TransferToRenounced {});
+    }
+
+    OWNERSHIP.update(deps.storage, |ownership| {
+        // Check sender and verify governance is not immutable
+        check_ownership_can_change(&deps.querier, &ownership, sender)?;
+        // NOTE: We don't validate the expiry, i.e. asserting it is later than
+        // the current block time.
+        //
+        // This is because if the owner submits an invalid expiry, it won't have
+        // any negative effect - it's just that the pending owner won't be able
+        // to accept the ownership.
+        //
+        // By not doing the check, we save a little bit of gas.
+        //
+        // To fix the erorr, the owner can simply invoke `transfer_ownership`
+        // again with the correct expiry and overwrite the invalid one.
+
+        let new_ownership = if let GovernanceDetails::NFT { .. } = new_owner {
+            // if NFT, no way for NFT contract to accept so just confirm
+            Ownership {
+                owner: new_owner,
+                pending_owner: None,
+                pending_expiry: None,
+            }
+        } else {
+            Ownership {
+                pending_owner: Some(new_owner),
+                pending_expiry: expiry,
+                ..ownership
+            }
+        };
```

Comment:
@CyberHoward This auto-accept doesn't feel right to me. NFT contract cannot accept it, but we still can get owner of it.

We have 3 endpoints on UpdateOwnership

- Transfer ownership: propose owner
- Accept ownership: proposed governance accepts ownership
- Renounce: toss ownership to void

And this NFT case doesn't match to the regular usage of `Transfer Ownership`, so I would suggest either put it into pending_owner and check owner of NFT during accept, or create separate endpoint `NFT: Toss ownership to nft`

Reply:
Implemented first suggestion, IMO that's preferred way to handle it

Reply:
Ok so you're suggesting that the owner of the NFT must accept the ownership for this so succeed? That makes sense.

Reply:
> Ok so you're suggesting that the owner of the NFT must accept the ownership for this so succeed? That makes sense.

Yes, nice!